### ISSUE TITLE
How can I train my own dataset?

### ISSUE THREAD

**Issue by shonehang** [isht7/pytorch-deeplab-resnet#22]
Thank you for sharing this code, I can train VOC database by your code.

Issue 1：I can train VOC，but it stoped at 17636 when I set max iter is 40000. I want to know the reason,
![1727430532](https://user-images.githubusercontent.com/11374747/33611667-2e18076c-da0a-11e7-89e8-2ca1ac655406.jpg)

Issue 2：I want to train my own dataset, which has 7 classes. But the model dict size of the init model  is for 21 classes. How can I train this network without init model using my dataset?

Thank you very much! Look forward to your reply! @isht7

**Comment by isht7** [comment]
The error you show seems to be because data_list has been exhausted. You can modify [this](https://github.com/isht7/pytorch-deeplab-resnet/blob/master/train.py#L205) and change 10 to 20 to remove this error. For training network with 7 classes, read the Readme, updates section, second point under 24 June 2017.

**Comment by isht7** [comment]
The first error shouldn't have occurred. Maybe you are not training on the full set of images.

**Comment by shonehang** [comment]
Thank you very much!!! I have solved my problem according to your reply.
Yes, I should change the line you pointed. Because my maxIter=40000, batch_size=6 and the number of train_aug=10582, so 40000 × 6 / 10582 ≈ 23, I changed like this: for i in range(23) and everything will be OK!
Now I can train my own dataset too, my train dataset is 176  images of 1000×1000 size. I set scale=1.0, batch_size=1, maxIter=30000, I must zoom my images to be 890 × 890 and the memory usage is 11787MB(Titan X). If the image size becomes larger, it will be out of memory, so now I only can train images of 890 × 890 size.
Thank you~
